### DISCUSSION TITLE
How to save user Name when loggin in with Magic Link?

### DISCUSSION THREAD

**Discussion by sarkarology** [supabase/supabase#9292] [Questions]
I am trying the below code:

```
const signInWithEmail = async(e) => {
    e.preventDefault();
    try {
      const { user, error } = await supabase.auth.signIn({
          email: e.target.email.value,
        },
        {
          data: {
            name: e.target.email.name
          }
        }
      )
      if (error) throw error
    } catch(error) {
      alert(error.error_description || error.message)
    } finally {
    }
  }
```
The user successfully logs in via magic link sent to his email address. But I can't find the "name" of the user in the session or user objects.

Please help 🙏

**Comment by GaryAustin1** [comment]
Only SignUp takes the data parameter in v1.
In v2, it appears there is code to take the options:data parameter for signInWithOtp, but it is not documented and I've not tested it.

**Reply by sarkarology** [reply]
> Only SignUp takes the data parameter in v1. In v2, it appears there is code to take the options:data parameter for signInWithOtp, but it is not documented and I've not tested it.

Thanks, Gary. 

PS. I frequently see your replies here. Appreciate your effort to help others.
